packages/api/src/controllers/user.ts

```diff
@@ -140,7 +140,7 @@ async function getOrCreateSubscription(
     customer: stripeCustomerId,
   });
   if (existing.data.length > 0) {
-    return existing.data[0];
+    return null;
```

Comment:
This was the only bit I wasn't sure about - I'm not sure what the correct way to control flow in this way is in Typescript

Reply:
I think this function should stay as a "get or create". The idea I think is that a user should always have a subscription on the stripe side, but only one. So this func is idempotent and after being called we have that guarantee and the existing or new subscription is returned.

If you want to hide fields from the subscription in the response instead, WDYT of creating a function like [`cleanUserFields`](https://github.com/livepeer/livepeer-com/blob/8891a083255bd13ffa4f4b10422e71a267da2231/packages/api/src/controllers/user.ts#L48)? Or perhaps not even return anything (204) if we don't need it in the frontend. 

Would only need to change [this bit](https://github.com/livepeer/livepeer-com/blob/8891a083255bd13ffa4f4b10422e71a267da2231/packages/www/hooks/use-api/endpoints/user.ts#L343) I guess? The `subscription` object is returned as the response in some apis here

Reply:
Right, I don't think we have any flows calling this where we'd expect the user to have an existing subscription though and so I thought it might be cleaner to explicitly mark this as a "create" until we need that.

Reply:
Oh I see. I have never touched this code myself so I don't really know what I'm talking about lol

I thought the flow was sth like:
 - Registration: `POST /user` always creates a subscription with the default product id (free), so every user always has a subscription, it just might be the free one. Here it would be a create.
 - When the user changes their plan, `POST /create-subscription` is called, which has a misleading name and actually _updates_ the existing subscription, doesn't actually create one. Here it would be a get.

I haven't checked the frontend code to know what API it calls when user changes plan tho, I just assumed that partt.

If that's not the case then LGTM!

Reply:
Good catch, it looks like I'd missed it because we actually call "/register" again when changing plan 🤦